cocotb/share/makefiles/Makefile.pylib.Msys

```diff
@@ -54,5 +54,30 @@ empty :=
 PYTHON_VERSION=$(subst $(period),$(empty),$(LOCAL_PYTHON_VERSION))
 PYTHON_INCLUDEDIR := $(shell $(PYTHON_BIN) -c 'import distutils.sysconfig; print( distutils.sysconfig.get_python_inc() )')
 
-PYLIBS = -lPython$(PYTHON_VERSION)
-PYTHON_DYN_LIB = Python$(PYTHON_VERSION).dll
+SYSCONFIG_LDLIBRARY = $(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; print( sysconfig.get_config_var("LDLIBRARY") )')
+
+ifeq ($(SYSCONFIG_LDLIBRARY),None)
+    PYLIBS := -lPython$(PYTHON_VERSION)
+else
+    PYLIBS:=$(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; import os; print("-l"+os.path.splitext(sysconfig.get_config_var("LDLIBRARY"))[0][3:])')
+endif
+
+# Some simulator accept only paths in form of "c:/../.."
+ifneq ($(VERILOG_SOURCES),)
+    VERILOG_SOURCES := $(shell cygpath -m $(VERILOG_SOURCES))
+endif
+ifneq ($(VHDL_SOURCES),)
+    VHDL_SOURCES := $(shell cygpath -m $(VHDL_SOURCES))
+endif
+
+PWD ?= $(shell pwd)
+
+# sys.path in form "/c/../:/c/.."
+PYTHON_SYS_PATH := $(shell $(PYTHON_BIN) -c "import sys, os; print(':'.join(['/'+dir.replace('\\\ ',' ').replace(os.sep,'/').replace(':','') for dir in sys.path]))")
+PYTHONPATH_ESCAPE := $(shell echo "$(PYTHON_SYS_PATH)" | sed 's/(/\\(/g' | sed 's/)/\\)/g' | sed 's/ /\\ /g')
+PYTHONPATH := $(PYTHONPATH_ESCAPE)
+
+PYTHONHOME:=$(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; print( sysconfig.get_config_var("prefix") )')
+PATH_ESCAPE := $(shell echo "$(PATH)" | sed 's/(/\\(/g' | sed 's/)/\\)/g' | sed 's/ /\\ /g')
+LIB_LOAD := PATH=$(LIB_DIR):$(PYTHONPATH_ESCAPE):$(PATH_ESCAPE) PYTHONHOME=$(PYTHONHOME)
```

Comment:
Why are we putting `sys.path` in `PATH`? The python include path and the system binary path are very different things, aren't they.

Reply:
This is a good question but it does not work other way.